lib/source-map-consumer.js

```diff
@@ -199,146 +195,134 @@ exports.SourceMapConsumer = SourceMapConsumer;
  *
  * [0]: https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?pli=1#
  */
-function BasicSourceMapConsumer(aSourceMap, aSourceMapURL) {
-  let sourceMap = aSourceMap;
-  if (typeof aSourceMap === "string") {
-    sourceMap = util.parseSourceMapInput(aSourceMap);
-  }
-
-  const version = util.getArg(sourceMap, "version");
-  let sources = util.getArg(sourceMap, "sources");
-  // Sass 3.3 leaves out the 'names' array, so we deviate from the spec (which
-  // requires the array) to play nice here.
-  const names = util.getArg(sourceMap, "names", []);
-  let sourceRoot = util.getArg(sourceMap, "sourceRoot", null);
-  const sourcesContent = util.getArg(sourceMap, "sourcesContent", null);
-  const mappings = util.getArg(sourceMap, "mappings");
-  const file = util.getArg(sourceMap, "file", null);
-
-  // Once again, Sass deviates from the spec and supplies the version as a
-  // string rather than a number, so we use loose equality checking here.
-  if (version != this._version) {
-    throw new Error("Unsupported version: " + version);
-  }
-
-  if (sourceRoot) {
-    sourceRoot = util.normalize(sourceRoot);
-  }
+class BasicSourceMapConsumer extends SourceMapConsumer {
+  constructor(aSourceMap, aSourceMapURL) {
+    return super(INTERNAL).then(that => {
```

Comment:
You thought `that` was a holdover from before arrow functions.  But in this case, the constructor returns a Promise, which is pretty non-standard.  So hence the workaround.